extra/test/baremetal_eventrouter_test.cc

```diff
@@ -1 +1,149 @@
+                    &MockModule<Module::D>::m_module,
+                },
+            .m_num_modules = 4,
+        },
+    };
+    const ErOptions_t m_options{
+        .m_tasks     = m_tasks,
+        .m_num_tasks = 1,
+        .m_IsInIsr   = nullptr,
+    };
+};
+
+}  // namespace
+
+namespace testing
+{
+
+class ErBaremetalTest : public Test
+{
+   protected:
+    ErBaremetalTest() { ErInit(&m_options.m_options); }
+    ~ErBaremetalTest()
+    {
+        ErNewLoop();
+        assert(ErGetEventToDeliver() == nullptr);
+        ErDeinit();
+    }
+
+    MockOptions m_options;
+};
+
+TEST_F(ErBaremetalTest, SendEventAndReuse)
+{
+    constexpr int kSendingModule     = MockOptions::Module::A;
+    constexpr int kSubscribingModule = MockOptions::Module::B;
+
+    ErEvent_t event = {
+        .m_type            = ER_EVENT_TYPE__FIRST,
+        .m_reference_count = {0},
+        .m_sending_module  = &MockModule<kSendingModule>::m_module,
+        .m_next            = {.m_next = NULL},
+    };
+
+    ErSubscribe(&MockModule<kSubscribingModule>::m_module, event.m_type);
+
+    ErSend(&event);
+
+    MockModule<kSubscribingModule>::m_event_handler_ret =
+        ER_EVENT_HANDLER_RET__HANDLED;
+
+    ErNewLoop();
+    EXPECT_EQ(ErGetEventToDeliver(), &event);
+    ErCallHandlers(&event);
+    EXPECT_EQ(MockModule<kSendingModule>::m_last_event_handled, &event);
+    EXPECT_EQ(ErGetEventToDeliver(), nullptr);
+
+    ErSend(&event);
+
+    ErNewLoop();
+    EXPECT_EQ(ErGetEventToDeliver(), &event);
+    ErCallHandlers(&event);
+    EXPECT_EQ(MockModule<kSendingModule>::m_last_event_handled, &event);
+}
+
+TEST_F(ErBaremetalTest, SendTwoEventsAndReuse)
+{
+    constexpr int kSendingModule     = MockOptions::Module::A;
+    constexpr int kSubscribingModule = MockOptions::Module::B;
+
+    ErEvent_t event1 = {
+        .m_type            = ER_EVENT_TYPE__1,
+        .m_reference_count = {0},
+        .m_sending_module  = &MockModule<kSendingModule>::m_module,
+        .m_next            = {.m_next = NULL},
+    };
+
+    ErEvent_t event2 = {
+        .m_type            = ER_EVENT_TYPE__2,
+        .m_reference_count = {0},
+        .m_sending_module  = &MockModule<kSendingModule>::m_module,
+        .m_next            = {.m_next = NULL},
+    };
+
+    ErSubscribe(&MockModule<kSubscribingModule>::m_module, event1.m_type);
+    ErSubscribe(&MockModule<kSubscribingModule>::m_module, event2.m_type);
+
+    ErSend(&event1);
+    ErSend(&event2);
+
+    MockModule<kSubscribingModule>::m_event_handler_ret =
+        ER_EVENT_HANDLER_RET__HANDLED;
+
+    ErNewLoop();
+    EXPECT_EQ(ErGetEventToDeliver(), &event1);
+    ErCallHandlers(&event1);
+    EXPECT_EQ(MockModule<kSendingModule>::m_last_event_handled, &event1);
+    EXPECT_EQ(ErGetEventToDeliver(), &event2);
+    ErCallHandlers(&event2);
+    EXPECT_EQ(MockModule<kSendingModule>::m_last_event_handled, &event2);
+    EXPECT_EQ(ErGetEventToDeliver(), nullptr);
```

Comment:
Shouldn't these be delivered to `kSubscribingModule` here?

Reply:
Oh, Yes. Both `kSubscribingModule` (for subscribe) and `kSendingModule` (for return to sender) should be delivered.
I'll add the test codes. Thank you!!

Reply:
I updated the test code and all tests are passed.